barcode_library_characterizer/library_quality_control.py

```diff
@@ -135,13 +135,50 @@ def build_hamming_distance_multithreading(
             row[1]
             for row in reader ]
 
-    pairs = [ (slow,fast) for slow in range(len(barcodes))
-              for fast in range(slow+1, len(barcodes)) ]
-
-    for slow,fast in pairs:
-        distance = hamming_distance(barcodes[fast], barcodes[slow])
-        hamming_histogram[distance] += 1
+    return build_histogram_simple(barcodes)
+        
+        
+def build_histogram_simple(barcodes):
+    arr = [0]*(len(barcodes[0])+1)
+    c = Counter( hamming_distance(barcodes[fast], barcodes[slow])
+                    for slow,fast in pairs(len(barcodes)) )
+    for i,count in c.items():

```

Comment:
Is this only needed to reformat the Counter into an array to be able to compare when testing?

Reply:
Correct, just for comparison during testing. But I suspect that representing the result as a Vector with component wise addition is the correct representation for an implementation that's easily parallelizable, see the `mapReduce` case. 

I suspect that using a numpy array might be a good start